### DISCUSSION TITLE
Mantle / Vault through Window that the character would fit through?

### DISCUSSION THREAD

**Discussion by xkamronx** [Sixze/ALS-Refactored#412] [Ideas]
I'm poking through the Actions.cpp and wondering how difficult it would be to add vaulting through a window that the character would logically fit through?

**Comment by MrWilsonWilson** [comment]
I am going to start doing the same thing for my job, I think ths logic is all there, you only need to check on a low mantle if the space behind the obstacle is free to vault to, or if you should only mantle and cclimb on it.
Difficulty is that it's all done and replicated in cpp so you need to be comfortable on it, I'm more of a bp nerd but I'll take on the challenge.

**Reply by xkamronx** [reply]
Did you ever end up doing this?

**Comment by colorindarkness** [comment]
You'll also need to check if the object is less than a Maximum Depth. You'll also need a new Root Motion Source,

**Comment by MrWilsonWilson** [comment]
I ended up doing an implementation im not really proud of, it works and
does not break, does some checks on depths and capsule size on arrival and
whatnot, but I was never getting the "snap orientation and hands on the
right spot and move feeling", the root motion source used for the mantling
does a lot of calculations based on Z axis checks difference and whatnot,
where a vaulting actually can end up with same Z. So it's complex, doable
but complex!

**Reply by xkamronx** [reply]
I noticed it can cause issues if using an actual vault animation in the same manner and kind of teleports the character all over the place due to the root motion not being a straight line like the default mantle animations, but I'm getting closer to something that I feel matches the original mantle's quality as far as getting hands on the right spot, the turn towards the wall, etc
